src/VisualStudio/IntegrationTest/New.IntegrationTests/CSharp/CSharpGoToImplementation.cs

```diff
@@ -66,12 +61,10 @@ public async Task SimpleGoToImplementation(bool asyncNavigation)
                 results[0].NavigateTo(isPreview: false, shouldActivate: true);
 
                 await TestServices.Workarounds.WaitForNavigationAsync(HangMitigatingCancellationToken);
-
-                identifierWithCaret = "$$Implementation";
```

Comment:
❗ This is a significant change in behavior. It means the Find All References results window no longer places the caret at the beginning of an identifier when you select an item. I don't believe this is a desirable change, so we should review it before moving in this direction.

Reply:
Why was it ok before w/o async navigation?

Reply:
@jasonmalinowski  @CyrusNajmabadi 

There used to be a difference in navigation when we found a single item in GTD vs, when the user clicks on FAR/GTD result. In the former case we selected the target span and place cursor at the end. In the latter we did not select anything and placed cursor at the start if the identifier.

This change unifies both cases. As implemented it places the cursor at the end of the identifier, but we can also place it at the start.

Any idea why GTD with a single item selected the span and placed cursor at the end? Would it be ok to change that to place at the start w/o selection?

Reply:
Tracked it down to 
```
textManager.NavigateToLineAndColumn2(
    vsTextBuffer,
    VSConstants.LOGVIEWID.TextView_guid,
    vsTextSpan.iStartLine,
    vsTextSpan.iStartIndex,
    vsTextSpan.iEndLine,
    vsTextSpan.iEndIndex,
    (uint)_VIEWFRAMETYPE.vftCodeWindow));
```

This implements the selection + cursor at the end behavior.

Interactive Window uses TextView API:

```C#
textView.Selection.Select(surfaceBufferSpan.Start, surfaceBufferSpan.End);
textView.ViewScroller.EnsureSpanVisible(surfaceBufferSpan.SnapshotSpan, EnsureSpanVisibleOptions.AlwaysCenter);
textView.Caret.MoveTo(surfaceBufferSpan.Start);

textView.VisualElement.Focus();
```

It selects the span and places cursor at the start. It seems like that's what we would want everywhere.